internal/acctest/template/generator_sdk.go

```diff
@@ -0,0 +1,93 @@
+package template
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// SDKTemplateGenerator creates Go templates for Terraform SDK resources
+type SDKTemplateGenerator struct {
+	extractor *SDKFieldExtractor
+	renderer  *CommonTemplateRenderer
+}
+
+// NewSDKTemplateGenerator creates a new template generator for SDK resources
+func NewSDKTemplateGenerator() *SDKTemplateGenerator {
+	return &SDKTemplateGenerator{
+		extractor: NewSDKFieldExtractor(),
+		renderer:  &CommonTemplateRenderer{},
+	}
+}
+
+// GenerateTemplate implements the TemplateGenerator interface
+func (g *SDKTemplateGenerator) GenerateTemplate(schemaObj interface{}, resourceType string, kind ResourceKind) string {
+	// For SDK resources, we expect a *schema.Resource
+	resource, ok := schemaObj.(*schema.Resource)
+	if !ok {
```

Comment:
nit: looks like returning an error explicitly might be beneficial. We call this inly in one place from the `Store`, where we have access to T and can fail the test right away.